### PR TITLE
Platoon: Waypoint management improvements

### PR DESCRIPTION
**Pull request by gabryelreyes**
- Send Waypoints in constant interval (distance)
- Fixed feedback topic
- At least 2 Waypoints required to start driving

### CODE REVIEW

File: lib/PlatoonService/src/PlatoonController.cpp [hunk 1]
```diff
@@ -157,8 +155,26 @@ void PlatoonController::process()
...
+    }
+
+    /* Are there enough waypoints available at the start of the drive? */
+    if ((false == m_processingChainRelease) && (MIN_AVAILABLE_WAYPOINTS < numberOfAvailableWaypoints))
+    {
+        m_processingChainRelease = true;
+    }
     /* Process chain on timeout. */
-    if ((true == m_processingChainTimer.isTimeout()) && (nullptr != m_motorSetpointCallback))
+    else if ((true == m_processingChainTimer.isTimeout()) && (nullptr != m_motorSetpointCallback))
```
Comment (BlueAndi) [inline lib/PlatoonService/src/PlatoonController.cpp:177]: Why checking for m_motorSetpointCallback? The comment talks only about the timeout.

File: lib/PlatoonService/src/PlatoonController.h [hunk 2]
```diff
@@ -133,6 +135,16 @@ class PlatoonController
...
 
+    /**
+     * Distance interval between waypoints in mm.
+     */
+    static const int32_t WAYPOINT_DISTANCE_INTERVAL = 50;
+
+    /**
+     * Minimum number of available waypoints for release of processing chain.
+     */
+    static const size_t MIN_AVAILABLE_WAYPOINTS = 2;
```
Comment (BlueAndi) [inline lib/PlatoonService/src/PlatoonController.h:146]: 'U' is missing.

File: lib/PlatoonService/src/PlatoonController.h [hunk 3]
```diff
@@ -133,6 +135,16 @@ class PlatoonController
      */
     static const int32_t FORWARD_CONE_APERTURE = 1300; /* Aprox. 1/3 * pi */
```
Comment (BlueAndi) [inline lib/PlatoonService/src/PlatoonController.h:136]: Thats not 1/3 PI, isn't it?

File: lib/PlatoonService/src/PlatoonController.h [hunk 4]
```diff
@@ -133,6 +135,16 @@ class PlatoonController
...
 
+    /**
+     * Distance interval between waypoints in mm.
+     */
+    static const int32_t WAYPOINT_DISTANCE_INTERVAL = 50;
```
Comment (BlueAndi) [inline lib/PlatoonService/src/PlatoonController.h:141]: Can it be negative or is the calculation later with signed data types?

**Review by BlueAndi — CHANGES_REQUESTED** [review]

File: lib/PlatoonService/src/PlatoonController.h (hunk 4 above)
Comment (gabryelreyes) [reply lib/PlatoonService/src/PlatoonController.h:141]: Calculation is later with signed data types (PlatoonController.cpp:159)

**Review by BlueAndi — APPROVED** [review]

**Pull request merged**